test/controllers/ht_users_controller_test.rb

```diff
@@ -254,13 +254,14 @@ def setup
 
 class HTUsersControllerCSVTest < ActionDispatch::IntegrationTest
   def setup
-    @inst1 = create(:ht_institution, entityID: "http://example.com", inst_id: "X", name: "Y")
+    @inst1 = create(:ht_institution, entityID: "http://counterexample.com",
+      inst_id: "X", name: "Y")
     @user1 = HTUser.new(userid: "a@b", displayname: "A B", email: "c@d",
       activitycontact: "e@f", approver: "g@h",
       authorizer: "i@j", usertype: "staff", role: "ssd",
       access: "total", expires: "2020-01-01 00:00:00",
       expire_type: "expiresannually", iprestrict: "any",
-      mfa: false, identity_provider: "http://example.com")
+      mfa: false, identity_provider: "http://example.com", inst_id: "X")
```

Comment:
Not sure I understand 1) why the user still lists an `identity_provider` and 2) why this test passes...

Reply:
> I can't find any uses in catalog, ls, or pt. [Searching the organization](https://github.com/search?q=org%3Ahathitrust+identity_provider&type=code) gives a few hits. Most of these, I think, are getting the data from the Shib session.
> 
> One that definitely uses that column is a direct SQL call in [mdp-lib ACL logic](https://github.com/hathitrust/mdp-lib/blob/0053cb982db7b91215eb7b6a5ff9daee8c202c5e/Auth/ACL.pm#L592-L604). It was added by @respinos five years ago and is used if `$do_restrict_to_identity_provider` is true.

`$do_restrict_to_identity_provider` is an obsolete mechanism of ensuring a friend account couldn't appear as one of the institutions that was using email addresses as the identifier. (We decided not to do this.)

Reply:
It's a roundabout way of testing the join, with `example.com` changed to `counterexample.com`. This test would fail if the code were still insisting on a join via identity provider. This is one of the tests that initially failed with the new join, and for whatever reason I didn't bother to create a more explicit test after tweaking it.

Reply:
I added an explicit test in `models/ht_users_test.rb` and dialed back the unnecessary stuff here by removing the institution's `entityID` as irrelevant. That should make this test a little less head-scratchy.